cosmos/db.go

```diff
@@ -11,18 +11,9 @@ import (
 
 // wrapRecoverError recover error from panic and wrap it into a db error.
 func wrapRecoverError(err *error) {
-	if *err != nil {
-		return
+	if r := recover(); *err == nil && r != nil {
+		*err = fmt.Errorf("db: %s", r)
```

Comment:
i really want to wrap the error into `fmt.Errorf("db: %w", errR)`. that why the code was a bit long